downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/Biopax.java

```diff
@@ -0,0 +1,173 @@
+	  	releaseNumber = releaseNumber.replace("/", "");
+	  	String biopaxDir = releaseNumber + "_biopax";
+        File biopaxDirFile = new File(biopaxDir);
+        if (!biopaxDirFile.exists()) {
+        	biopaxDirFile.mkdir();
+        }
+        // First, generate owl files. This particular step requires a local maven installation of the PathwayExchange jar
+		for (int i = 0; i < 2; i++) {
+			int biopaxLevel = i + 2;
+			if (i == 0) {
+				logger.info("Generating BioPAX2 Species owl Files...");
+				SpeciesAllPathwaysConverter converter = new SpeciesAllPathwaysConverter();
+				converter.doDump(new String[]{host, database, username, password, port, biopaxDir});
+			} else {
+				logger.info("Generating BioPAX3 Species owl Files...");
+				SpeciesAllPathwaysLevel3Converter level3converter = new SpeciesAllPathwaysLevel3Converter();
+				level3converter.doDump(new String[]{host, database, username, password, port, biopaxDir});
+			}
+			
+			// Rename files, replacing whitespace with underscores
+		    Files.newDirectoryStream(Paths.get(biopaxDir),
+		    	      path -> path.toString().endsWith(".owl"))
+		    	      .forEach(f -> {
+		    	          String spath = f.toFile().getPath();
+		    	          spath = spath.replaceAll(" +", "_");
+		    	          File formattedFile = new File(spath);
+		    	          f.toFile().renameTo(formattedFile);
+		    	      });
+		    // Validate each owl file in the biopax output directory produced by PathwaysConverter
+		    logger.info("Validating owl files...");
+			runBiopaxValidation(biopaxDir);
+
+			// Compress all Biopax and validation files into individual zip files
+			logger.info("Zipping BioPAX" + biopaxLevel + " files...");
+			FileOutputStream biopaxOutputStream;
+			FileOutputStream validatorOutputStream;
+			if (i == 0) {
+				biopaxOutputStream = new FileOutputStream("biopax2.zip");
+				validatorOutputStream = new FileOutputStream("biopax2_validator.zip");
+			} else {
+				biopaxOutputStream = new FileOutputStream("biopax.zip");
+				validatorOutputStream = new FileOutputStream("biopax_validator.zip");
+			}
+			ZipOutputStream biopaxZipStream = new ZipOutputStream(biopaxOutputStream);
+			ZipOutputStream validatorZipStream = new ZipOutputStream(validatorOutputStream);
+		    Files.newDirectoryStream(Paths.get(biopaxDir),
+    	      path -> (path.toString().endsWith(".xml") || path.toString().endsWith(".owl")))
+    	      .forEach(f -> {
+    	          String spath = f.toFile().getPath();
+    	          File file = new File(spath);
+	        	  try {
+	    	          if (spath.endsWith(".owl")) {
+	    	        	writeToZipFile(file, biopaxZipStream);
+	    	          } else if (spath.endsWith(".xml")) {
+						writeToZipFile(file, validatorZipStream);
+	    	          }
+				  } catch (IOException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
+				  }
+    	      });
+		    biopaxZipStream.close();
+		    validatorZipStream.close();
```

Comment:
The zipping logic could all be extracted to its own method called after the logging statement